app/models/product.rb

```diff
@@ -35,4 +40,16 @@ def self.search(query)
     }
     __elasticsearch__.search(search_params).records
   end
+
+  def self.store_autocomplete_names
+    Product.pluck(:name).each do |name|
+      redis_instance = Redis.new
+      redis_instance.zadd('autocomplete_names', 0, name.downcase)
```

Comment:
Refactor this. We are plucking All products names every time when we create or update a product which is not necessary.